server/routerlicious/packages/routerlicious-base/src/riddler/tenantManager.ts

```diff
@@ -657,6 +674,46 @@ export class TenantManager {
 		return this.buildTenantConfig(updatedtenantDocument);
 	}
 
+	public async updatePublicNetworkAccessPolicy(
+		tenantId: string,
+		publicNetworkAccessEnabled: boolean,
+	): Promise<ITenantConfig> {
+		const tenantDocument = await this.getTenantDocument(tenantId);
+		if (tenantDocument === undefined) {
+			Lumberjack.error("Could not find tenantId.", {
+				[BaseTelemetryProperties.tenantId]: tenantId,
+			});
+			throw new NetworkError(404, `Could not find tenant: ${tenantId}`);
+		}
+
+		const updates: Partial<ITenantDocument> = {};
+		const isTenantPublicNetworkAccessEnabled =
+			this.getTenantPublicNetworkAccessEnabled(tenantDocument);
+
+		if (publicNetworkAccessEnabled === isTenantPublicNetworkAccessEnabled) {
+			// no update needed
+			return this.buildTenantConfig(tenantDocument);
+		}
+
+		updates.publicNetworkAccessEnabled = publicNetworkAccessEnabled;
+
+		await this.runWithDatabaseRequestCounter(async () =>
+			this.tenantRepository.update({ _id: tenantId }, updates, null),
+		);
+
+		const updatedtenantDocument = await this.getTenantDocument(tenantId);
```

Comment:
Just a thought on second-glance: from our experience with the createDoc flow and having to add a couple retries on 404s due to DB propagation, is it posisble that this could have a similar issue?

The other problem could be because of different DB connections making the check, so it might be irrelevant for back-to-back calls on the same connection. Just want to call it out. Maybe something to keep an eye on